StepWat_R_Wrapper_Parallel/STEPWAT.Wrapper.MAIN_V3.R

```diff
@@ -47,13 +46,374 @@ tick_on<-proc.time()
+
+#print a warning if there are no rgroup inputs for this site
+if(nrow(rgroup_data_site) == 0 & !contains_vector){
+  print(paste("Site",site,"contains no rgroup inputs.", sep = " "))
+}
+
+#Get all treatments pertaining to site or "all"
+treatments<-unique(rgroup_data_site$treatment)
+
+#For each treatment for the site in question generate a rgroup.in file
+for(i in treatments)
+{
+  df=rgroup_data_site[rgroup_data_site$treatment==i,]
+  #Get rid of Site and treatment columns
+  df <- subset(df, select = -c(1,2) )
+  
+  #Populate the dist.freq vector with fire frequency inputs
+  temp<-df['killfrq']
+  temp<-unique(temp)
+  temp<-as.numeric(temp)
+  dist.freq.current<-temp
+  
+  #Populate the graz.freq vector with grazing frequency inputs
+  temp<-df['grazing_frq']
+  temp<-unique(temp)
+  temp<-as.numeric(temp)
+  graz.freq.current<-temp
+  
+  #If grazing is ocurring we need to know the level.
+  if(graz.freq.current != 0){
+    #Populate the graz_intensity vector with grazing intensity inputs
+    temp<-df['proportion_grazing']
+    temp<-unique(temp)
+    temp<-max(temp)
+  } else {
+    temp="0"
+  }
+  graz_intensity.current<-temp
+  
+  # If any of the inputs contain a decimal like "0.24" this code will remove the "0." and leave the "24"
+  # this is necessary for functionallity in wrapper.file
+  dist.freq.current <- toString(dist.freq.current)
+  graz.freq.current <- toString(graz.freq.current)
+  graz_intensity.current <- toString(graz_intensity.current)
+  
+  dist.freq.current <- strsplit(dist.freq.current,"\\.")
+  graz.freq.current <- strsplit(graz.freq.current,"\\.")
+  graz_intensity.current <- strsplit(graz_intensity.current,"\\.")
+        
+  dist.freq.current <- dist.freq.current[[1]][length(dist.freq.current[[1]])]
+  graz.freq.current <- graz.freq.current[[1]][length(graz.freq.current[[1]])]
+  graz_intensity.current <- graz_intensity.current[[1]][length(graz_intensity.current[[1]])]
+        
+  # Now add the file name to the list of file names
+  rgroups <- c(rgroups, paste0("rgroup.","freq.",dist.freq.current,".graz.",graz.freq.current,".",graz_intensity.current,".",i))
+        
+  # write the file
+  write.table(df, file = paste0("rgroup.","freq.",dist.freq.current,".graz.",graz.freq.current,".",graz_intensity.current,".",i,".in"),quote=FALSE,row.names=FALSE,col.names = FALSE,sep="\t")
+}
+
+#Store the files names in the rgroup_files variable
+rgroup_files<-list.files(path=".",pattern = "rgroup")
+rgroup_files<-rgroup_files[rgroup_files!="rgroup_template.in"]
+
+#Append rgroup_template.in withing STEPWAT_DIST to all the created files and save with a unique filename
+for (i in rgroup_files)
+{
+  system(paste("cat ","rgroup_template.in >>",i,sep=""))
+}
+
+setwd("..")
+
+########################### Garbage collection for species, rgroup, and soil requirements #######################
+# All of these variables were created in the species, soils, and rgroup sections. They will never be used again.
```

Comment:
I will add better documentation to this section to explain why we removed these variables, as well as add a comment letting future programmers know that they should delete their variables as soon as they are no longer needed.